File: Number_Theory_Functions/nthFibonacci/nthFibonacci.java
// Java Implementation
public class FibonacciFormula {

    public static long nthFibonacci(int n) {
        /**
         * Returns the n-th Fibonacci number using Binet's formula.
         */
        double phi = (1 + Math.sqrt(5)) / 2;
        double psi = (1 - Math.sqrt(5)) / 2;
        return Math.round((Math.pow(phi, n) - Math.pow(psi, n)) / Math.sqrt(5));
    }

    public static void main(String[] args) {
        // Example use cases
        int[] testCases = {0, 1, 2, 5, 10, 20, 50};
        for (int n : testCases) {
            System.out.println("F(" + n + ") = " + nthFibonacci(n));
        }
    }
}

File: Number_Theory_Functions/nthFibonacci/nthFibonacci.py
# Python Implementation
import math

def nthfibonacci(n):
    """
    Returns the n-th Fibonacci number using Binet's formula.
    """
    phi = (1 + math.sqrt(5)) / 2
    psi = (1 - math.sqrt(5)) / 2
    return round((phi**n - psi**n) / math.sqrt(5))

if __name__ == "__main__":
    # Example use cases
    test_cases = [0, 1, 2, 5, 10, 20, 50]
    for n in test_cases:
        print(f"F({n}) = {nthfibonacci(n)}")
